pandas/tests/indexes/test_base.py

```diff
@@ -340,14 +340,17 @@ def test_constructor_empty_special(self, empty, klass):
         "index",
         [
             "datetime",
-            "float",
-            "int",
+            "float64",
+            "float32",
+            "int64",
+            "int32",
```

Comment:
just for my understanding, why don't `'int16'` and `'int8'` need to be here as well? (likewise for other tests)

this isn't a request to change it, just trying to understand

Reply:
I thought it was a bit strange to test these things here, and suspect these should be located in `pandas.tests.indexes.common.Base` and adapted in its subclasses as needed. At the same time I didn't really want to delve too deep into refactoring the code base, as it would distract from my main objective right now i.e. getting rid of the old numeric indexes. I probably shouldn't even have added the 32bit versions, but I was curious if it would still pass.

How about I write an issue about this refactoring and fix it after I've removed Int64Index & friends?

Reply:
yeah no objections

Reply:
Thanks, I've written a issue, see #49679.